### PR TITLE
feat: add support for sfu migration

### PR DESCRIPTION
**Pull request by gulzar1996**
### Details(context, link the issue, how was the bug fixed, what does the new feature do)

-
-

### CODE REVIEW

File: examples/prebuilt-react-integration/src/App.jsx [hunk 1]
```diff
@@ -9,5 +9,16 @@ export default function App() {
...
 
-  return <HMSPrebuilt roomCode={roomCode} />;
+  return (
+    <HMSPrebuilt
+      roomCode={roomCode}
+      options={{
+        endpoints: {
+          tokenByRoomCode: 'https://auth-nonprod.100ms.live/v2/token',
+          roomLayout: 'https://api-nonprod.100ms.live/v2/layouts/ui',
+          init: 'https://qa-in2-ipv6.100ms.live/init',
```
Comment (eswarclynn) [inline examples/prebuilt-react-integration/src/App.jsx:19]: Intentional for testing?

File: packages/hms-video-store/src/notification-manager/NotificationManager.ts [hunk 2]
```diff
@@ -168,6 +169,11 @@ export class NotificationManager {
         this.policyChangeManager.handlePolicyChange(notification as PolicyParams);
         break;
 
+      case HMSNotificationMethod.NODE_INFO:
+        console.log({ notification });
```
Comment (eswarclynn) [inline packages/hms-video-store/src/notification-manager/NotificationManager.ts:173]: Needed?

File: packages/hms-video-store/src/transport/index.ts [hunk 3]
```diff
@@ -159,6 +172,7 @@ export default class HMSTransport {
     onTrickle: async (trickle: HMSTrickle) => {
       const connection =
         trickle.target === HMSConnectionRole.Publish ? this.publishConnection : this.subscribeConnection;
+      console.log({ trickle, connection });
```
Comment (KaustubhKumar05) [inline packages/hms-video-store/src/transport/index.ts:175]: Can be removed?

File: packages/hms-video-store/src/transport/index.ts [hunk 4]
```diff
@@ -591,6 +421,96 @@ export default class HMSTransport {
...
+    }
+    localPeer.auxiliaryTracks = auxTracks;
+    tracksToPublish = tracksToPublish.concat(auxTracks);
+    streamMap.clear();
+    for (const track of tracksToPublish) {
+      await this.publishTrack(track);
+      this.listener?.onTrackUpdate(HMSTrackUpdate.TRACK_ADDED, track, localPeer);
+    }
+  }
+
```
Comment (KaustubhKumar05) [inline packages/hms-video-store/src/transport/index.ts:513]: If this happens during screenshare, we should stop screenshare with the older track ID? And handle the UI

File: packages/hms-video-store/src/transport/index.ts [hunk 5]
```diff
@@ -716,20 +650,199 @@ export default class HMSTransport {
...
+          const log = newState === 'disconnected' ? HMSLogger.w.bind(HMSLogger) : HMSLogger.d.bind(HMSLogger);
+          log(TAG, `Publish connection state change: ${newState}`);
+          if (newState === 'new') {
+            return;
+          }
+
+          if (newState === 'connected') {
+            this.connectivityListener?.onICESuccess(true);
+            this.publishConnection?.handleSelectedIceCandidatePairs();
+          } else if (newState === 'failed') {
```
Comment (KaustubhKumar05) [inline packages/hms-video-store/src/transport/index.ts:720]: Should we use enums for states?

File: packages/hms-video-store/src/transport/index.ts [hunk 6]
```diff
@@ -716,20 +650,199 @@ export default class HMSTransport {
...
+          console.log(`onIceCandidate: ${candidate}`);
+          this.connectivityListener?.onICECandidate(candidate, false);
+        },
+
+        onSelectedCandidatePairChange: candidatePair => {
+          this.connectivityListener?.onSelectedICECandidatePairChange(candidatePair, false);
+        },
+      };
       if (!this.publishConnection) {
+        console.log('creating publish connection');
```
Comment (KaustubhKumar05) [inline packages/hms-video-store/src/transport/index.ts:831]: Should we remove this or use HMSLogger?

File: examples/prebuilt-react-integration/src/App.jsx (hunk 1 above)
Comment (raviteja83) [reply examples/prebuilt-react-integration/src/App.jsx:19]: testing will remove.

File: packages/hms-video-store/src/notification-manager/NotificationManager.ts (hunk 2 above)
Comment (raviteja83) [reply packages/hms-video-store/src/notification-manager/NotificationManager.ts:173]: can remove

File: packages/hms-video-store/src/transport/index.ts (hunk 4 above)
Comment (raviteja83) [reply packages/hms-video-store/src/transport/index.ts:513]: yes, remove track remove is sent above. it is handled.

File: packages/hms-video-store/src/transport/index.ts (hunk 5 above)
Comment (raviteja83) [reply packages/hms-video-store/src/transport/index.ts:720]: it will auto complete. fixed values are there.

**Review by KaustubhKumar05 — APPROVED** [review]

File: packages/hms-video-store/src/transport/index.ts [hunk 7]
```diff
@@ -1081,15 +1189,7 @@ export default class HMSTransport {
...
     if (this.publishConnection) {
-      const p = new Promise<boolean>((resolve, reject) => {
-        this.callbacks.set(RENEGOTIATION_CALLBACK_ID, {
-          promise: { resolve, reject },
-          action: HMSAction.RESTART_ICE,
-          extra: {},
-        });
-      });
       await this.performPublishRenegotiation({ iceRestart: this.publishConnection.connectionState !== 'connected' });
-      await p;
```
Comment (eswarclynn) [inline packages/hms-video-store/src/transport/index.ts:1092]: Is this not required anymore?

**Review by eswarclynn — APPROVED** [review]

File: packages/hms-video-store/src/transport/index.ts (hunk 7 above)
Comment (raviteja83) [reply packages/hms-video-store/src/transport/index.ts:1092]: not required as the above funciton resolves.

File: packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts [hunk 8]
```diff
@@ -70,7 +70,7 @@ export class HMSLocalVideoTrack extends HMSVideoTrack {
     source: string,
     private eventBus: EventBus,
     settings: HMSVideoTrackSettings = new HMSVideoTrackSettingsBuilder().build(),
-    room?: Room,
+    private room?: Room,
```
Comment (KaustubhKumar05) [inline packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts:73]: Why was this updated?

**Comment by KaustubhKumar05** [comment]
In the next testing session can we test with AV plugins as well @raviteja83?

File: packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts (hunk 8 above)
Comment (raviteja83) [reply packages/hms-video-store/src/media/tracks/HMSLocalVideoTrack.ts:73]: to be able to clone

**Review by raviteja83 — APPROVED** [review]

**Review by KaustubhKumar05 — APPROVED** [review]

**Pull request merged**